### ISSUE TITLE
Prompt doesn't always show even if setAlwaysShow(true) is called

### ISSUE THREAD

**Issue by jiawenzhang** [stkent/amplify#139]
tested on 1.2.0

if screen orients, prompt shows up again.

**Comment by jiawenzhang** [comment]
tested with 1.2.0

**Comment by stkent** [comment]
Thanks for the report! I'm a little confused about the expected/desired behavior vs the actual behavior here. Could you please elaborate with more details and steps to reproduce?

**Comment by jiawenzhang** [comment]
Here is my setup:

in my main Application.java

```
Amplify.initSharedInstance(this)
                 .setFeedbackEmailAddress("XXX")
                 .setAlwaysShow(true)
```

Then in one of my activities, prompt is triggered by:

`Amplify.getSharedInstance().promptIfReady(mPromptView);`

the prompt shows up the first time promptIfReady is called, but does not show the second time it is called. On the second time, however, when the screen orientation changes, the prompt suddenly shows up.

Is setAlwaysShow(true) supposed to show the prompt each time promptIfReady is called? This used to be the behaviour in 1.0.0.

I am testing 1.2.0 with a custom prompt view on Nexus 4, Android 4.4.

**Comment by stkent** [comment]
Can you please show the code where `Amplify.getSharedInstance().promptIfReady(mPromptView);` is called? Also, are you using a `CustomLayoutPromptView` instance or your own implementation of `IPromptView`?

**Comment by jiawenzhang** [comment]
in my activity's onCreate I have this:

```
mPromptView = (CustomLayoutPromptView) findViewById(R.id.feedback_prompt_view);
        mPromptView.addPromptEventListener(new IEventListener() {
            @Override
            public void notifyEventTriggered(@NonNull final IEvent event) {
                if (event == PromptViewEvent.PROMPT_SHOWN) {
                    Analytics.send(Analytics.APP, "Feedback", "Show");
                } else if (event == PromptViewEvent.PROMPT_DISMISSED) {
                    Analytics.send(Analytics.APP, "Feedback", "Dismiss");
                }
            }
        });
```

and in the same activity's onActivityResult, I have
```
protected void onActivityResult(int requestCode, int resultCode, Intent data) {
...
    Amplify.getSharedInstance().promptIfReady(mPromptView);
...
}
```

so the prompt is triggered when another activity gets dismissed and returned to this activity.

I am using CustomLayoutPromptView, as you can see from the code.

**Comment by stkent** [comment]
Hmm, interesting. I have a couple ideas as to what this might be caused by. Please confirm I'm understanding the flow of events correctly:

- First `Activity` is on-screen.
- First `Activity` launches a second `Activity` for a result.
- Second `Activity` finishes and returns a result to the first `Activity`.
- **Expected**: prompt shows. **Actual**: prompt shows.
- User does not interact with prompt at all.
- First `Activity` launches a second `Activity` for a result.
- Second `Activity` finishes and returns a result to the first `Activity`.
- **Expected**: prompt shows. **Actual**: prompt does not show.
- User rotates device.
- Prompt suddenly shows.

If you turn on debug logging (by calling `.setLogLevel(Logger.LogLevel.DEBUG)` in your custom `Application` class), does the output suggest that the prompt is being blocked due to a tracked event?

**Comment by jiawenzhang** [comment]
"User does not interact with prompt at all" should be "User dismisses the
prompt" in the flow
all other events are correct.

I set Logger.LogLevel.DEBUG, but did not see any log.

**Comment by stkent** [comment]
Ah, ok, then I'm pretty sure [this line](https://github.com/stkent/amplify/blob/1.2.0/amplify/src/main/java/com/github/stkent/amplify/prompt/BasePromptView.java#L187) is the issue. As a temporary fix, try updating your `onActivityResult` as follows:

```java
protected void onActivityResult(int requestCode, int resultCode, Intent data) {
...
    Amplify.getSharedInstance().promptIfReady(mPromptView);
    mPromptView.setVisibility(View.VISIBLE);
...
}
```

If that works then I will add a fix in release 1.4.0 of the library, which should be completed this coming week.

**Comment by stkent** [comment]
@jiawenzhang please try v1.3.0 of the library and see if that resolves your issue. Thanks!
